app/Models/Photo.php

```diff
@@ -278,75 +279,75 @@ public function city ()
      */
     public function smoking ()
     {
-    	return $this->hasOne('App\Models\Litter\Categories\Smoking', 'id', 'smoking_id');
+    	return $this->belongsTo('App\Models\Litter\Categories\Smoking', 'smoking_id', 'id');
```

Comment:
A photo has one category : smoking ?

Reply:
Using `hasOne` doesn't work well when retrieving things like `$photo->smoking`, I noticed that when I was refactoring into actions, those lines with `$photo->fresh()->$category()->create()`. The relationship type should reflect the way foreign keys are placed between tables, and, in our case, since a photo has a smoking_id, the photo belongs to smoking. HasOne would make sense if smoking had a photo_id.

Anyway, the database design could be improved in my opinion. The right way to model the relationship photo - category would be a [Many to Many](https://laravel.com/docs/8.x/eloquent-relationships#many-to-many). This is true because a photo can be related to many categories, and a category would be applied to many photos.
This would result in a new table, called photos_categories, which would connect the two types of models. In turn, we would have a separate table for `tags -> (category_id, tag_name)`. This way we'd have a more normalized database, and it would be easier to make changes, and if I'm not wrong is the reason why we're migrating to Redis, right?

Reply:
The belongTo now makes sense to me. Nice find. I like the sound of the new photos_categories and tags table! I think we will keep redis mostly for metadata like total_photos, leaderboards, and keep the photos + tags data on mysql